### ISSUE TITLE
Wrong colorization in Yaml files

### ISSUE THREAD

**Issue by sugdyzhekov** [wholroyd/vscode-jinja#3]
Could you check following examples? Looks like Jinja colorization works wrong.

Example 1:
![example1](https://cloud.githubusercontent.com/assets/8955225/16642881/e27a2d32-4430-11e6-914e-5faf4d57e9aa.png)

``` yaml
setup_plesk:
  cmd.run:
    - name: |
        PSA_PASSWORD={{ pass }} /usr/sbin/plesk bin init_conf --init -shared_ips add:{{ ip }} -license_agreed true -admin_info_not_required true -send_announce false -send_tech_announce false -hostname {{ fqdn }} -passwd "" -company plesk.ru -name Administrator -email {{ email }}
        ...
```

Example 2:
![example2](https://cloud.githubusercontent.com/assets/8955225/16642880/e260e0e8-4430-11e6-93b3-14ef737b1a05.png)

``` yaml
setup_plesk:
  cmd.run:
    - name: PSA_PASSWORD={{ pass }} /usr/sbin/plesk bin init_conf --init -shared_ips add:{{ ip }} -license_agreed true -admin_info_not_required true -send_announce false -send_tech_announce false -hostname {{ fqdn }} -passwd "" -company plesk.ru -name Administrator -email {{ email }}
```

Example 3:
![example3](https://cloud.githubusercontent.com/assets/8955225/16642837/88d2576e-4430-11e6-8139-a50a9f33c9d0.png)

``` yaml
create_default_subscription:
  cmd.run:
    - name: /usr/sbin/plesk bin subscription -c {{ fqdn }} -hosting true -ip {{ ip }} -login support -passwd $(/usr/bin/openssl rand -base64 12)
    - unless: /usr/sbin/plesk bin subscription -i {{ fqdn }} &>/dev/null
```

**Comment by wholroyd** [comment]
Thanks for letting me know about the issue.

The way VS Code handles languages is different depending on if they are core to the editor or come as an extension. Extensions can't override core languages very easily and have to use hackery sometimes to get things to work.

Give me a couple of days to take a look at it to see how to implement the changes.

**Comment by wholroyd** [comment]
To expand a bit on my last comment, see https://github.com/Microsoft/vscode/issues/1751 for more information on what I mean.

I can't immediately fix this issue without rewriting the YAML syntax from scratch or until Microsoft fixes the language override issue. This will take some time to fix, but I can do it.

**Comment by wholroyd** [comment]
At this point, I don't see myself being able to rewrite the YAML language to make this work and this is technically not a bug with the plugin, but a limitation of Code. As such, there is nothing I can really do about this until I get time or they fix language overrides in Code. I'm closing as Won't Fix for now.

**Comment by sugdyzhekov** [comment]
Ok, I got it. Thank you for your efforts!

чт, 10 нояб. 2016 г. в 9:51, William Holroyd notifications@github.com:

**Comment by wholroyd** [comment]
Wanted to let you know that VSCode added support for embedded and injected languages recently. How does this look?

![image](https://cloud.githubusercontent.com/assets/147990/20487648/329b8768-afd2-11e6-8f0c-dd5be3a732fd.png)

**Comment by wholroyd** [comment]
Ended up getting blocked on this as it didn't work in YAML for all scenarios, so it's on pause.

Linking to https://github.com/wholroyd/vscode-jinja/issues/19.

**Comment by wholroyd** [comment]
YAML+Jinja is Jinja YAML (.yml.j2, .sls, etc.). Built-in YAML is left alone. #23.
